src/data/helper/dimensionHelper.ts

```diff
@@ -130,6 +131,9 @@ export function summarizeDimensions(
                 getOrCreateEncodeArr(userOutputEncode, coordDim)[coordDimIndex] =
                     data.getDimensionIndex(dimItem.name);
             }
+            else {
+                lastExtraCoord = dimName;
```

Comment:
The [strategy A](https://github.com/apache/echarts/pull/15851/commits/ddd5f9e113bbb6ec6ec20d5ad9545fba24184d25#r729775267)  means that: 

> (A) if no defaultedLabel selected, use the last !isExtraCoord dimension no matter it's category or time

That is, find the last coord that is "not extra coord".
But here the code finds the last coord that is "extra coord".